plugins/sentinelone/komand_sentinelone/tasks/monitor_activities_and_events/task.py

```diff
@@ -0,0 +1,337 @@
+                    lookback_timestamp,
+                    activities_logs_next_cursor,
+                )
+                all_logs.extend(activities_logs)
+            else:
+                self.logger.info("Pagination in progress, skipping activities log collection...")
+        if queries.get(EVENTS_LOGS):
+            events_logs_next_cursor = cursors.get(EVENTS_LOGS)
+            if is_paginating is False or (is_paginating is True and events_logs_next_cursor):
+                self.logger.info("Getting device control events logs...")
+                events_logs = self.get_generic_logs(
+                    EVENTS_LOGS, state, total_forbidden_responses, lookback_timestamp, events_logs_next_cursor
+                )
+                all_logs.extend(events_logs)
+            else:
+                self.logger.info("Pagination in progress, skipping events log collection...")
+        if queries.get(THREATS_LOGS):
+            threats_logs_next_cursor = cursors.get(THREATS_LOGS)
+            if is_paginating is False or (is_paginating is True and threats_logs_next_cursor):
+                self.logger.info("Getting threats logs...")
+                threats_logs = self.get_generic_logs(
+                    THREATS_LOGS, state, total_forbidden_responses, lookback_timestamp, threats_logs_next_cursor
+                )
+                all_logs.extend(threats_logs)
+            else:
+                self.logger.info("Pagination in progress, skipping threats log collection...")
+        return all_logs, total_forbidden_responses
+
+    def get_generic_logs(
+        self,
+        log_type: str,
+        state: dict,
+        total_forbidden_responses: int,
+        last_run_timestamp: str = None,
+        pagination_cursor: str = None,
+    ) -> List[Dict]:
+        """
+        Determine log type and query that log set with query parameters determined by type, cursor, and last timestamp.
+        Set latest log timestamp and page cursor in state.
+        Return logs from query
+        """
+        timestamp_key = LOG_TYPE_MAP.get(log_type, {}).get("LAST_LOG_TIMESTAMP")
+        page_token_key = LOG_TYPE_MAP.get(log_type, {}).get("PAGE_TOKEN")
+        last_log_timestamp = state.get(timestamp_key)
+        if not last_log_timestamp:
+            last_log_timestamp = STARTING_TIMESTAMP
+        query_params = self.get_query_params(log_type, last_run_timestamp, pagination_cursor)
+        if log_type == ACTIVITIES_LOGS:
+            response = self.connection.client.get_activities_list(query_params, True, False)
+        elif log_type == EVENTS_LOGS:
+            response = self.connection.client.get_device_control_events(query_params, True, False)
+        else:
+            response = self.connection.client.get_threats(query_params, API_VERSION, True, False)
+        try:
+            status_code = response.status_code
+            self.logger.info(f"Received {status_code} for {log_type} query")
```

Comment:
could be another logger we could toggle debug/info?

Reply:
I like logging out the status code, mainly just to log out 403, 401 errors. I've added a condition just for this. What do you think?

Reply:
Makes sense to log for 403/401 errors, just thought it were logging success codes it would be a lot of logs. Having the condition sounds good 😄